sandpack-react/src/components/CodeEditor/CodeMirror.tsx

```diff
@@ -185,6 +195,54 @@ export const CodeMirror: React.FC<CodeMirrorProps> = ({
     }
   }, [code]);
 
+  React.useEffect(
+    function messageToInlineError() {
+      if (showInlineErrors) {
+        const unsubscribe = listen((message) => {
+          const view = cmView.current;
+
+          if (message.type === "success") {
+            view?.dispatch({
+              // Pass message to clean up inline error
+              annotations: [
+                ({
+                  type: "clean-error",
+                  value: null,
+                } as unknown) as Annotation<unknown>,
+              ],
+
+              // Trigger a doc change to remove inline error
+              changes: {
```

Comment:
question: isn't the annotation dispatch enough to trigger the change in the editor? seems like an extra operation here. don't see a big problem with it as it is not that often, but still, wondering if it couldn't introduce rendering problems in the future

Reply:
Unfortunately, the annotation isn't enough to clean up the old decorations, that's why it needs to trigger a new change to redraw all lines. This is the biggest downside of this implementation, but I wasn't able to figure out another way to do it.